lib/qif2json.js

```diff
@@ -114,35 +159,56 @@ exports.parse = function parse(qif, options) {
         break;
       case '$':
         division.amount = parseFloat(line.substring(1));
-        if (!(transaction.division instanceof Array)) {
-          transaction.division = [];
+        if (!(entity.division instanceof Array)) {
+          entity.division = [];
         }
-        transaction.division.push(division);
+        entity.division.push(division);
         division = {};
-
         break;
+      case '!':
+        const typeName = line.substring(1);
+        type = TYPES.find(({ name }) => name === typeName);
+        if (typeName === 'Account') {
+          isMultiAccount = true;
+        }
+        if (!type && typeName.startsWith('Type:')) {
+          type = {
+            type: typeName,
+            list_name: 'transactions',
+          };
+        }
+        if (isMultiAccount === false) {
+          data = appendEntity(data, { list_name: 'accounts' }, { type: typeName }, currentBankName, isMultiAccount);
+        }
 
+        if (!type) {
+          throw new Error(`File does not appear to be a valid qif file: ${line}. Type ${typeName} is not supported.`);
+        }
+        break;
       default:
+        // eslint-disable-next-line no-console
+        console.log(line);
```

Comment:
If you need this information then the error thrown on the next line should contain it, rather than console.logging

That, or use `debug()` for this (and add some context to explain why you're outputting it)

Reply:
I incorporated line into error message.